client/search/services/SearchReport.ts

```diff
@@ -0,0 +1,235 @@
+import {cloneDeep, isArray, isBoolean, pickBy} from 'lodash';
+import moment from 'moment';
+
+import {IArticle, IRestApiResponse} from 'superdesk-api';
+import {
+    IReportPayload,
+    IReportParams,
+    IStateFilter,
+    IDataFilter,
+    IArchiveStateCode,
+    IDateFilter,
+    DATA_FIELD,
+    ITEM_STATE,
+} from '../../interfaces';
+
+import {appConfig, apiQuery, superdeskApi} from '../../superdeskApi';
+
+class SearchReportService {
```

Comment:
It's definitely a good move to convert angular services to plain js/ts.

I'd suggest to not overuse classes too much. If there are pure functions in the class, like `getSupportedDataFields` or `getSupportedItemStates`, it's better to move their definition out of the class. It's easier to understand the code that way.

When you see a function in the root scope, you can be sure it's not using any state from the class. With class methods, every method can use internal class state and call all other methods so you have to analyze usages to understand the boundaries of every method.

Reply:
Sure, I will move private functions out. The reason I chose a class is so I only needed to change DI to an import, and the rest of the files stay the same.

Would you prefer I make all these functions and use "import search report from '....'"?

Reply:
Yep, I would make them all independent functions and would export an object containing the functions like:

```
function getSupportedItemStates() { /* ... */ }
function getSupportedDataFields() { /* ... */ }

export const searchReportService = {
    getSupportedItemStates,
    getSupportedDataFields,
};
```

It's possible to export them individually, but it's helpful to get an autocomplete when using services so an object seems a better choice to me.

Reply:
Good stuff I'll do that tomorrow. It's also easier to mock in tests when exporting as an object